packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+  FacetState,
+} from '@coveo/headless';
+import {
+  Component,
+  h,
+  Listen,
+  State,
+  Element,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Button} from '../../../common/button';
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
+import {InitializePopoverEvent, popoverClass} from './popover-type';
+import {FacetInfo} from '../../../common/facets/facet-common-store';
+
+interface ClearPopoverEvent {
+  ownId?: string;
+}
+
+/**
+ * @internal
+ * The `atomic-popover` component displays any facet as a popover menu.
+ *
+ * @slot default - The required slotted facet.
+ * @part popover-button - The button to click to display or hide the popover menu.
+ * @part label - The associated facet label.
+ * @part value-count - Number of selected values for the facet
+ * @part arrow-icon - The arrow icon on the button to display or hide the popover menu.
+ * @part placeholder - The placeholder displayed when the facet is loading.
+ * @part slot-wrapper - The wrapper that contains the 'facet' or 'slot'.
+ */
+@Component({
+  tag: 'atomic-popover',
+  styleUrl: 'atomic-popover.pcss',
+  shadow: true,
+})
+export class AtomicPopover implements InitializableComponent {
+  @Element() host!: HTMLElement;
+  private hasValues?: () => boolean;
+  private numberOfSelectedValues?: () => number;
```

Comment:
Instead of spreading the facet methods & attributes over multiple attributes, I would prefer if they were part of a single object, like `@State() childFacet`.